File: spring-boot-rest-services-all-examples-for-springboottutorial.com/src/main/java/com/in28minutes/springboot/configuration/BasicConfiguration.java
package com.in28minutes.springboot.configuration;

import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.stereotype.Component;

@Component
@ConfigurationProperties("basic")
public class BasicConfiguration {
	private boolean value;
	private String message;
	private int number;

	public boolean isValue() {
		return value;
	}

	public void setValue(boolean value) {
		this.value = value;
	}

	public String getMessage() {
		return message;
	}

	public void setMessage(String message) {
		this.message = message;
	}

	public int getNumber() {
		return number;
	}

	public void setNumber(int number) {
		this.number = number;
	}
}

File: spring-boot-kotlin-basics-configuration/src/main/kotlin/com/in28minutes/springboot/tutorial/basics/application/configuration/BasicConfiguration.kt
package com.in28minutes.springboot.tutorial.basics.application.configuration

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.stereotype.Component

@Component
@ConfigurationProperties("basic")
class BasicConfiguration {
    var isValue = false
    var message: String? = null
    var number = 0
}